api/api/result.go

```diff
@@ -313,3 +313,74 @@ func (r *Result) Get(ctx *gin.Context) {
 
 	ctx.Data(http.StatusOK, "application/octet-stream", result)
 }
+
+func (r *Result) Status(ctx *gin.Context) {
+	pid := ctx.Param("pid")
+	/*
+	 * There's an interesting timing issue here - if /result is called before
+	 * the job is scheduled and the header written, it is considered pending.
+	 *
+	 * The fact that the token checks out means that it is essentially pending
+	 * - it's enqueued, but no processing has started [1]. Also, partial
+	 * results have a fairly short expiration set, and requests to /result
+	 * after expiration would still carry a valid auth token.
+	 *
+	 * The fix here is probably to include created-at and expiration in the
+	 * token as well - if the token checks out, but the header does not exist,
+	 * the status is pending.
+	 *
+	 * [1] the header-write step not completed, to be precise
+	 */
+	body, err := r.Storage.Get(headerkey(pid)).Bytes()
+	if err == redis.Nil {
+		/* request sucessful, but key does not exist */
+		ctx.JSON(http.StatusAccepted, gin.H {
+			"location": fmt.Sprintf("result/%s/status", pid),
+			"status": "pending",
+		})
+		return
+	}
+	if err != nil {
+		log.Printf("Unable to get process header: %v", err)
+		ctx.AbortWithStatus(http.StatusNotFound)
```

Comment:
Should this perhaps be internal server error?

Reply:
Yes.